### ISSUE TITLE
Add Type Hints

### ISSUE THREAD

**Issue by 2e0byo** [tamland/python-tidal#137]
Part of the intention of dropping support for python 2.7 was being able to use type hints.  This would help with https://github.com/tehkillerbee/mopidy-tidal/issues/112 as well.

**Comment by 2e0byo** [comment]
[self-assigned but feel free to do it for me ;) ]

**Comment by arusahni** [comment]
Would you accept this as a series of smaller PRs, or would you prefer it land in one fell swoop? I figure incrementally introducing typing will be easier, but that's based on a few minutes of poking around as I worked with the API.

**Comment by 2e0byo** [comment]
> Would you accept this as a series of smaller PRs?

Definitely.  I'd do it in one swoop if I sat down to do it, but anything helps.  In my experience 80-90% of type hinting is low hanging fruit, and then you hit the remaining code where you either have to do something inventive or consider tweaking the implementation.  So if that turns out to be the case here, we should do the low hanging fruit first so people can start benefitting.

**Comment by arusahni** [comment]
Thanks! I put up an initial PR to get the baseline stuff out of the way (type checking, config, etc.). Once we're in agreement, I can start landing the bigger pieces of API surface.
